doc/source/toolchain.rst

```diff
@@ -71,9 +71,10 @@ The table shows the NumPy versions suitable for each major Python version
 =================  ========================    ===========================
 2.7 (SciPy 1.2)      1.8.2                      1.16.x
 3.5 (SciPy 1.4)      1.13.3                     1.18.x
-3.6                  1.14.5                     >= 1.18.x
-3.7                  1.14.5                     >= 1.18.x
+3.6 (SciPy 1.5)      1.14.5                     >= 1.18.x
+3.7                  1.16.5                     >= 1.18.x
 3.8                  1.17.3                     >= 1.18.x
+3.9                  1.19.3                     >= 1.18.x
```

Comment:
The maximum versions here could also uniformly move to `>= 1.20.x`...

Reply:
Further down under `C Compilers`, python 3.9 should also be added (not necessarily in this PR)

Reply:
```suggestion
3.7                  1.16.5                     >= 1.20.x
3.8                  1.17.3                     >= 1.20.x
3.9                  1.19.3                     >= 1.20.x
```

Suggestion for updated numpy maximum versions for those python versions still supported by 1.20.

Reply:
makes sense, adopted